packages/controls/src/widget_dragdrop.ts

```diff
@@ -0,0 +1,205 @@
+
+import * as _ from 'underscore';
+
+export
+class DraggableBoxModel extends CoreDOMWidgetModel {
+    defaults(): Backbone.ObjectHash {
+        return _.extend(super.defaults(), {
+            _view_name: 'DraggableBoxView',
+            _model_name: 'DraggableBoxModel',
+            child: null,
+            draggable: true,
+            drag_data: {}
+        });
+    }
+
+    static serializers = {
+        ...CoreDOMWidgetModel.serializers,
+        child: {deserialize: unpack_models}
+    };
+}
+
+export
+class DropBoxModel extends CoreDOMWidgetModel {
+    defaults(): Backbone.ObjectHash {
+        return _.extend(super.defaults(), {
+            _view_name: 'DropBoxView',
+            _model_name: 'DropBoxModel',
+            child: null
+        });
+    }
+
+    static serializers = {
+        ...CoreDOMWidgetModel.serializers,
+        child: {deserialize: unpack_models}
+    };
+}
+
+class DragDropBoxViewBase extends DOMWidgetView {
+    child_view: DOMWidgetView | null;
+    pWidget: JupyterLuminoPanelWidget;
+
+    _createElement(tagName: string): HTMLElement {
+        this.pWidget = new JupyterLuminoPanelWidget({ view: this });
+        return this.pWidget.node;
+    }
+
+    _setElement(el: HTMLElement): void {
+        if (this.el || el !== this.pWidget.node) {
+            // Boxes don't allow setting the element beyond the initial creation.
+            throw new Error('Cannot reset the DOM element.');
+        }
+
+        this.el = this.pWidget.node;
+        this.$el = $(this.pWidget.node);
+    }
+
+    initialize(parameters: WidgetView.IInitializeParameters): void {
+        super.initialize(parameters);
+        this.add_child_model(this.model.get('child'));
+        this.listenTo(this.model, 'change:child', this.update_child);
+
+        this.pWidget.addClass('jupyter-widgets');
+        this.pWidget.addClass('widget-container');
+        this.pWidget.addClass('widget-draggable-box');
+    }
+
+    add_child_model(model: WidgetModel): Promise<DOMWidgetView> {
+        return this.create_child_view(model).then((view: DOMWidgetView) => {
+            if (this.child_view && this.child_view !== null) {
+                this.child_view.remove();
+            }
+            this.pWidget.addWidget(view.pWidget);
+            this.child_view = view;
+            return view;
+        }).catch(reject('Could not add child view to box', true));
+    }
+
+    update_child(): void {
+        this.add_child_model(this.model.get('child'));
+    }
+
+    render(): void {
+        super.render();
+    }
+
+    remove(): void {
+        this.child_view = null;
+        super.remove();
+    }
+}
+
+export
+class DraggableBoxView extends DragDropBoxViewBase {
+    /** 
+     * Draggable mixin.
+     * Allows the widget to be draggable
+     *
+     * Note: In order to use it, you will need to add
+     * handlers for dragstartevent in the view class
+     * also need to call dragSetup at initialization time
```

Comment:
I'm not sure I understand this paragraph.

Also, I think we normally follow typedoc documentation and put API documentation above what it's documenting?

Reply:
Thanks for looking through this. Indeed, I haven't gotten around to update the docs! We don't have a mixin anymore, just a "simple" widget. Will change that :)

Reply:
I removed this outdated docs, and refreshed the docs for Python.